shell/platform/common/cpp/flutter_platform_node_delegate.h

```diff
@@ -0,0 +1,141 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_COMMON_CPP_FLUTTER_PLATFORM_NODE_DELEGATE_H_
+#define FLUTTER_SHELL_PLATFORM_COMMON_CPP_FLUTTER_PLATFORM_NODE_DELEGATE_H_
+
+#include "flutter/shell/platform/embedder/embedder.h"
+
+#include "flutter/third_party/accessibility/ax/ax_event_generator.h"
+#include "flutter/third_party/accessibility/ax/platform/ax_platform_node_delegate_base.h"
+
+namespace flutter {
+
+typedef ui::AXNode::AXID AccessibilityNodeId;
+
+//------------------------------------------------------------------------------
+/// The platform node delegate to be used in accessibility bridge. This
+/// class is responsible for providing native accessibility object with
+/// appropriate information, such as accessibility label/value/bounds.
+///
+/// While most methods have default implementations and are ready to be used
+/// as-is, the subclasses must override the GetNativeViewAccessible to return
+/// native accessibility objects. To do that, subclasses should create and
+/// maintain AXPlatformNode[s] which delegate their accessibility attributes to
+/// this class.
+///
+/// For desktop platforms, subclasses also need to override the GetBoundsRect
+/// to apply window-to-screen transform.
+class FlutterPlatformNodeDelegate : public ui::AXPlatformNodeDelegateBase {
+ public:
+  //----------------------------------------------------------------------------
+  /// The required interface to be able to own the flutter platform node
+  /// delegate.
+  class OwnerBridge {
+   public:
+    ~OwnerBridge() = default;
+
+   protected:
+    friend class FlutterPlatformNodeDelegate;
```

Comment:
Shouldn't this be unnecessary since this is a nested class?

Reply:
I think it is still needed? https://stackoverflow.com/questions/3537969/nested-classes-access-to-protected-member-of-enclosing-class-from-a-nested-prot

I tried to compile without the friend declaration, and it complained about accessing protected method.